### PR TITLE
Switch to new smart accounts sdk

### PR DESCRIPTION
**Pull request by jsidorenko**

### CODE REVIEW

File: src/services/smartWallet.js [hunk 1]
```diff
@@ -293,13 +294,15 @@ class SmartWallet {
...
-  getDeployEstimate(gasPrice: BigNumber) {
-    /**
-     * can also call `this.sdk.estimateAccountDeployment(REGULAR);`,
-     * but it needs sdk init and when migrating we don't have SDK initiated yet
-     * so we're using calculation method below that is provided by SDK creators
-     */
-    return utils.bigNumberify(790000).mul(gasPrice);
+  async estimateAccountDeployment(gasInfo: GasInfo) {
+    const deployEstimate = await this.sdk.estimateAccountDeployment().catch(() => {});
+    const gasAmount = get(deployEstimate, 'gasFee', 790000);
```
Comment (jsidorenko) [inline src/services/smartWallet.js]: `const gasAmount = get(deployEstimate, 'gasFee', new BigNumber(790000));`
to keep consistency

File: src/services/smartWallet.js [hunk 2]
```diff
@@ -171,7 +140,15 @@ class SmartWallet {
     const deployEstimate = await this.sdk.estimateAccountDeployment().catch(this.handleError);
 
     const accountBalance = this.getAccountRealBalance();
-    if (get(deployEstimate, 'totalCost') && accountBalance.gte(deployEstimate.totalCost)) {
+    if (get(deployEstimate, 'gasFee')
+      && get(deployEstimate, 'signedGasPrice.gasPrice')
+      && accountBalance.gte(deployEstimate.gasFee.mul(deployEstimate.signedGasPrice.gasPrice))
```
Comment (grilix) [inline src/services/smartWallet.js]: Would it make sense to extract this into a function?

Comment (grilix) [reply src/services/smartWallet.js]: Maybe it would just unnecessarily add extra complexity

Comment (poocart) [reply src/services/smartWallet.js]: Separated, please take a look and resolve if OK.

File: src/services/smartWallet.js (hunk 1 above)
Comment (poocart) [reply src/services/smartWallet.js]: Updated, please resolve if OK.

File: src/services/smartWallet.js [hunk 3]
```diff
@@ -56,6 +56,11 @@ type AccountTransaction = {
 
 let subscribedToEvents = false;
 
+const parseEstimatePayload = (estimatePayload) => ({
+  gasFee: get(estimatePayload, 'gasFee'),
+  gasPrice: get(estimatePayload, 'signedGasPrice.gasPrice'),
+});
```
Comment (jsidorenko) [inline src/services/smartWallet.js]: let's add total calculation here as well

File: src/services/smartWallet.js [hunk 4]
```diff
@@ -56,6 +56,11 @@ type AccountTransaction = {
 
 let subscribedToEvents = false;
 
+const parseEstimatePayload = (estimatePayload) => ({
+  gasFee: get(estimatePayload, 'gasFee'),
```
Comment (jsidorenko) [inline src/services/smartWallet.js]: maybe it's better to rename gasFee to gasAmount?

**Pull request merged**